### PR TITLE
Configure GELF logger

### PR DESCRIPTION
**Pull request by jmaupetit**
## Purpose

We need to send our logs to Graylog.

## Proposal

- [x] add the `gelf` log appender and make it configurable
- [x] add a Graylog stack to the project to test logs streaming

### CODE REVIEW

File: settings.json [hunk 1]
```diff
@@ -0,0 +1,465 @@
...
+   * Node native SSL support
+   *
+   * This is disabled by default.
+   * Make sure to have the minimum and correct file access permissions set so
+   * that the Etherpad server can access them
+   */
+
+  /*
+  "ssl" : {
+            "key"  : "/path-to-your/epl-server.key",
```
Comment (sampaccoud) [inline settings.json:117]: ?

Comment (jmaupetit) [reply settings.json:117]: :man_shrugging: this commented part is coming from [the original template](https://github.com/ether/etherpad-lite/blob/a817acbbccad9cbcc5fc69ccd9f4c8eafe36b86e/settings.json.docker#L117).

File: docker-compose.yml [hunk 2]
```diff
@@ -24,13 +32,58 @@ services:
...
+      - ./docker/files/usr/share/graylog/data/contentpacks:/usr/share/graylog/data/contentpacks
+    depends_on:
+      - mongo
+      - elasticsearch
+
+  # --- Extra tools
   dockerize:
     image: jwilder/dockerize
+
+  netshoot:
```
Comment (lunika) [inline docker-compose.yml:84]: You installed it for debugging purpose ? Is it intended to keep it ?

File: docker-compose.yml [hunk 3]
```diff
@@ -24,13 +32,58 @@ services:
...
+      GRAYLOG_ROOT_PASSWORD_SHA2: 8c6976e5b5410415bde908bd4dee15dfb167a9c873fc4bb8a81f6f2ab448a918
+      GRAYLOG_HTTP_EXTERNAL_URI: "http://127.0.0.1:9000/"
+      # Automatically create GELF UDP Input
+      GRAYLOG_CONTENT_PACKS_LOADER_ENABLED: "true"
+      GRAYLOG_CONTENT_PACKS_AUTO_INSTALL: gelf-udp-input.json
+    ports:
+      # Graylog web interface and REST API
+      - 9000:9000
+      # GELF TCP
+      - 12201:12201
```
Comment (lunika) [inline docker-compose.yml]: Is it useful to expose it ? Did you plan to use outside compose context ?

File: docker-compose.yml [hunk 4]
```diff
@@ -24,13 +32,58 @@ services:
...
+      # GELF UDP
+      - 12201:12201/udp
```
Comment (lunika) [inline docker-compose.yml]: Is it useful to expose it ? Did you plan to use outside compose context ?

File: docker-compose.yml (hunk 2 above)
Comment (jmaupetit) [reply docker-compose.yml:84]: As far as it is not started by default, I think it's worth it as a development tool (as `dockerize`).

File: docker-compose.yml (hunk 4 above)
Comment (jmaupetit) [reply docker-compose.yml]: Nope, you are right.

File: docker-compose.yml (hunk 2 above)
Comment (lunika) [reply docker-compose.yml:84]: We use dockerize in the makefile for example. If you think it's useful to keep it we can keep it.

Comment (jmaupetit) [reply docker-compose.yml:84]: I think it's useful as we have many services with many ports to handle. :pray:

File: docker/files/etc/nginx/conf.d/etherpad.conf [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+    proxy_set_header Host $host;
+    # recommended with keepalive connections
+    proxy_http_version 1.1;
+    # WebSocket proxying - from http://nginx.org/en/docs/http/websocket.html
+    proxy_set_header Upgrade $http_upgrade;
+    proxy_set_header Connection $connection_upgrade;
+    proxy_redirect off;
+
+    # Required to resolve the scaled backend application
+    resolver 127.0.0.11 valid=5s;
```
Comment (lunika) [inline docker/files/etc/nginx/conf.d/etherpad.conf:37]: Is this valid in a docker context ? Should not be `resolver etherpad` ? Using an ip address is a little bit strange don't you think ?

Comment (jmaupetit) [reply docker/files/etc/nginx/conf.d/etherpad.conf:37]: This is required for `nginx` to resolve the `etherpad` service IP in a docker-compose context. `127.0.0.11` is the IP of the docker network internal DNS.

Reference: https://nginx.org/en/docs/http/ngx_http_core_module.html#resolver

Comment (lunika) [reply docker/files/etc/nginx/conf.d/etherpad.conf:37]: :ok:

**Review by lunika — APPROVED** [review]

**Pull request merged**